android/titanium/src/java/org/appcelerator/titanium/io/TiResourceFile.java

```diff
@@ -31,34 +31,32 @@
 	private static final String TAG = "TiResourceFile";
 
 	private String path;
+	private String name;
 	private boolean statsFetched = false;
 	private boolean exists = false;
 
 	public TiResourceFile(String path)
 	{
 		super(TYPE_RESOURCE);
-		this.path = path;
+		this.path = (path != null) ? path : "";
+		fetchName();
 	}
 
 	@Override
 	public boolean isDirectory()
 	{
-		if (statsFetched) {
-			return this.exists && this.typeDir;
+		if (this.statsFetched == false) {
```

Comment:
Any particular reason there's a companions to false rather than typical `!this.statsFetched`?